File: 30zadStrongNumbersinInterval.js
let reminder = 0;
let sum = 0;
let m = 200;
let tmp = 0;

for(n = 1; n <= m; n++){
    tmp = n;
    while(tmp > 0){
        //shrink number
        reminder = tmp % 10;
        //factorial
        fact = 1;
        for(let i = 2; i <= reminder; i++){
            fact *= i
        }
        //add factorial to sum
        sum += fact
        tmp = Math.trunc(tmp/10)
    }

    //check sum
    if(sum == n){
        console.log(n + " is a strong num");
        sum = 0;
    }
    else{
        sum = 0;
    }
}

File: 30zadStrongNumbersInInterval.py
import math
reminder = 0
sum = 0
m = 200
tmp = 0

for n in range(1,m+1):
    tmp = n
    while n != 0:
        #shrink number
        reminder = n % 10
        #factorial
        fact = 1
        for i in range(2,reminder+1):
            fact *= i
        #add factorial to sum
        sum += fact
        n = math.trunc(n/10)
        
    #check sum
    if sum == tmp:
        print(tmp,' is a strong num')
        sum = 0
    else:
        sum = 0